fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionSpec.java

```diff
@@ -64,7 +97,7 @@ public boolean isSatisfy(DistributionSpec spec) {
                 return false;
             }
         }
-        return hashDistributionDesc.isSatisfy(other.hashDistributionDesc);
+        return hashDistributionDesc.isSatisfy(other.hashDistributionDesc) || isJoinEqColumnsCompatible(other);
```

Comment:
This looks like not or relation

Reply:
shuffle columns of `this` and `other` is different, so should be or relation

Reply:
Shuffle_Join[A,B] could satisfy the Shffle_join[A,B,C] ?

Reply:
updated